### PR TITLE
Remove PartitionStatefulSetStrategyType

### LINKED ISSUE #46975: StatefulSet Partitioned strategy vs parameterized Roll out strategy
Prior to 1.7 release we should decide on keeping the PartitionedStatefulSetStrategy documented in the design for StatefulSet update, or, instead, adding an optional ordinal parameter to the RollingUpdateStatefulSetStrategy. The deviation is described below.

```golang
// StatefulSetUpdateStrategy indicates the strategy that the StatefulSet
// controller will use to perform updates. It includes any additional parameters
// necessary to perform the update for the indicated strategy.
type StatefulSetUpdateStrategy struct {
	// Type indicates the type of the StatefulSetUpdateStrategy.
	Type StatefulSetUpdateStrategyType `json:"type,omitempty" protobuf:"bytes,1,opt,name=type,casttype=StatefulSetStrategyType"`
	// Partition is used to communicate the optional ordinal at which to partition
	// the StatefulSet when Type is RollingUpdateStatefulSetStrategyType. This
	// value may only be set when the Type is RollingUpdateStatefulSetStrategyType,
	// and it must be nil otherwise.
	Partition *RollingUpdatePartition `json:"partition,omitempty" protobuf:"bytes,2,opt,name=partition"`
}

// RollingUpdatePartition is used to indicate a partition in the StatefulSet during a Rolling Update.
type RollingUpdatePartition struct {
	// Ordinal indicates the ordinal at which the StatefulSet should be
	// partitioned. All Pods with an ordinal greater than or equal to partition will receive the update.
	Ordinal int32 `json:"ordinal" protobuf:"varint,1,opt,name=ordinal"`
}
```

### PR DESCRIPTION
**Pull request by kow3ns**
This PR removes PartitionStatefulSetStrategyType add adds a parameter to RollingUpdateStatefulSetStrategyType as described in the issue below. We need this PR to ensure that the StatefulSet API conforms to the existing API for DaemonSet.

fixes #46975
```release-note
NONE
```
@kargakis 
@smarterclayton 
@janetkuo

### CODE REVIEW

File: pkg/apis/apps/types.go [hunk 1]
```diff
@@ -100,12 +89,11 @@ const (
...
-// PartitionStatefulSetStrategy contains the parameters used with the
-// PartitionStatefulSetStrategyType.
-type PartitionStatefulSetStrategy struct {
-	// Ordinal indicates the ordinal at which the StatefulSet should be
+// RollingUpdateStatefulSetStrategy is used to communicate parameter for RollingUpdateStatefulSetStrategyType.
+type RollingUpdateStatefulSetStrategy struct {
+	// Partition indicates the ordinal at which the StatefulSet should be
 	// partitioned.
-	Ordinal int32
+	Partition int32 `json:"ordinal" protobuf:"varint,1,opt,name=ordinal"`
```
Comment (kargakis) [inline pkg/apis/apps/types.go]: Should be optional

Comment (kargakis) [inline pkg/apis/apps/types.go]: Ok, `rollingUpdate` is optional so this can stay required.

Comment (kargakis) [inline pkg/apis/apps/types.go]: Though, this is probably better as optional. If we add a new field in rollingUpdate and set it then we will require this field to be set, too, when we may not want such a thing.

**Comment by smarterclayton** [comment]
/approve

We cannot ship an API we don't intend to support.  The risk of breakage from removal is minimal

File: pkg/apis/apps/types.go (hunk 1 above)
Comment (kow3ns) [inline pkg/apis/apps/types.go]: Agreed, I've made it optional with a default of 0.

Comment (kargakis) [inline pkg/apis/apps/types.go]: I think until we add a new field, it's OK. Ultimately, I feel there is difference between 0 and nil (0 shouldn't update the status to signal a complete update but nil should so we can have post-hooks).

**Comment by kow3ns** [comment]
@kargakis Is this basically LGTM?

**Comment by kargakis** [comment]
Sorry, yes
/lgtm

**Pull request merged**